### PR TITLE
remove calling hook for enqueueing Site Editor assets

### PR DESCRIPTION
**Pull request by Addison-Stavlo**
#### Changes proposed in this Pull Request

Remove the `do_action` for enqueueing assets, as this was added into the site editor init function in core.

#### Testing instructions

* Build/Sync this FSE plugin along with D40583
* Smoke test the editors.

Fixes #

### CODE REVIEW

File: apps/full-site-editing/full-site-editing-plugin/posts-list-block/class-posts-list-block.php [hunk 1]
```diff
@@ -32,6 +32,7 @@ class Posts_List_Block {
 	private function __construct() {
 		add_action( 'init', array( $this, 'register_blocks' ), 100 );
 		add_action( 'enqueue_block_editor_assets', array( $this, 'enqueue_scripts' ), 100 );
+		add_action( 'enqueue_core_site_editor_assets', array( $this, 'enqueue_scripts' ), 100 );
```
Comment (vindl) [inline apps/full-site-editing/full-site-editing-plugin/posts-list-block/class-posts-list-block.php]: Interesting - I would expect this to be included via `enqueue_block_editor_assets`. Was that not the case?

**Comment by vindl** [comment]
Oh scratch that, the Site Editor was missing the `enqueue_block_editor_assets` hook in its init. Since it was introduced in core recently, the custom changes on our side are probably no longer needed.

See: https://github.com/WordPress/gutenberg/pull/20549/files#diff-5022ec1d311d9e1661b75192ce0385aaR195

**Comment by Addison-Stavlo** [comment]
> Based on latest core update I think we can just update this to remove our override function altogether.

Gotcha.  This makes sense.  I updated this PR to remove our override for calling the action.

It looks like those changes aren't on edge yet however, so we will have to put a pin in this until then.

**Comment by vindl** [comment]
> It looks like those changes aren't on edge yet however, so we will have to put a pin in this until then.

Yup, sounds good. 👍

**Pull request merged**